sources/editor/Stride.GameStudio/ViewModels/DebuggingViewModel.cs

```diff
@@ -228,6 +234,21 @@ private struct ModifiedAssembly
 
             public Project Project;
         }
+        /// <summary>
+        /// Blocks until condition is true or timeout occurs.
+        /// </summary>
+        /// <param name="condition">The break condition.</param>
+        /// <param name="frequency">The frequency at which the condition will be checked.</param>
+        /// <param name="timeout">The timeout in milliseconds.</param>
+        /// <returns></returns>
+        public static async Task WaitUntil(Func<bool> condition, int frequency = 25)
+        {
+            var waitTask = Task.Run(async () =>
+            {
+                while (!condition()) await Task.Delay(frequency);
+            });
+            await waitTask;
```

Comment:
Why running in another task here?

Reply:
https://github.com/stride3d/stride/pull/2106/commits/2ca2766a86b758a9b273040a6ad0cbf5be3e32a1
removed it, is this the way it should be?

Reply:
What's the goal here? To await for an asynchronous event? If so, there are other patterns that can be used instead of an active pull (which this repeated loop is a kind of).

For instance, `SemaphoreSlim` or a `TaskCompletionSource` could be used instead, depending on the expected usage.

Reply:
the goal here is to wait until a list is ready, the delay was introduced as i didnt think it would work without but it does
even without waiting it works as long as it fills up the dictionary and doesnt get blocked by the idispatcher

what i want to do is 
1. on first change spawn notification ( stride is blocked )
2. until the button is pressed sum up the assemblies in the dictionary
3. on button click take the dictionary and do the stuff that is allready there

the projectwatcher never completes his task, neither does the debugger complete his task

they run in parallel, mybe semaphore would work but i have no idea how to implement it as i just read quickly over it

Reply:
ok it was easier than i thought with asnyc enumerable
https://github.com/stride3d/stride/pull/2106/commits/3436b3f249625d7808a5b470cebd9c14226a53a8

problem is now i dont know what to write in the dispose... there was before a .Wait()... how can i actually stop that thing? or should it be stopped?

Reply:
got help from C# discord, i just have to run a break; to stop it